tests/commands/test_get.py

```diff
@@ -0,0 +1,432 @@
+@pytest.mark.parametrize('paths_expected', [
+    (['./one'], 4),
+    (['./one/two'], 3),
+    (['./one/two/three'], 2),
+    (['one/two/three'], 2),
+    (['/one/two/three'], 0),
+    (['./one/two/three/four'], 1),
+    (['./path/that/does/not/exist/but/is/very/long'], 0),
+    (['/path/that/does/not/exist/and/does/not/start/with/dot'], 0),
+    (['path/that/does/not/exist/and/does/not/start/with/dot/or/slash'], 0),
+    (['.', './one', './one/two/three/four'], 6),
+    (['./one/two/three/four', './another/dir'], 2),
+    (['def/ghi'], 0),
+    ([], 6)])
+def test_get_paths(repo: Repo, paths_expected: tuple[str, int]) -> None:
+    """
+    Test that `onyo get --path x` retrieves the expected assets.
+
+    Parametrized path contains Tuple([paths], n_assets expected to match).
+    """
+    paths, n_assets = paths_expected
+    cmd = ['onyo', 'get', '--depth', '999', '--path', *paths, '-H']
+    ret = subprocess.run(cmd, capture_output=True, text=True)
+    output = [output.split('\t') for output in ret.stdout.split('\n')][:-1]
+
+    # Ensure one of the given paths matches the outermost parents of each asset
+    paths = set(Path(path) for path in paths)
+
+    if paths:
+        for line in output:
+            assert any(paths & set(Path(line[-1]).parents))
+
+    assert len(output) == n_assets
+    assert not ret.stderr
+    assert ret.returncode == 0
+
+
+@pytest.mark.repo_contents(*convert_contents([
+    ('a13bc_foo_bar.1', {'num': 'num-3'}),
+    ('a2cd_foo_bar.2', {'num': 'num-16'}),
+    ('a36ab_foo_bar.3', {'num': 'num-20'})]))
+@pytest.mark.parametrize('sort', ['-s', '-S', ''])
+@pytest.mark.parametrize('keys', [
+    ['type'], ['num'], ['unset', 'type'], []])
+def test_get_sort(repo: Repo, sort: str, keys: list[str]) -> None:
+    """
+    Test that `onyo get --keys x y z` with `-s` (ascending) or `-S`
+    (descending)  retrieves assets in the expected 'natural sorted' order.
+    """
+    presorted = {
+        'path': ['a2cd_foo_bar.2', 'a13bc_foo_bar.1', 'a36ab_foo_bar.3'],
+        'type': ['a2cd', 'a13bc', 'a36ab'],
+        'num': ['num-3', 'num-16', 'num-20']}
+
+    cmd = ['onyo', 'get', '--depth', '999', '--keys', *keys, '-H']
+    cmd = cmd + [sort] if sort else cmd
+    ret = subprocess.run(cmd, capture_output=True, text=True)
+    output = [output.split('\t') for output in ret.stdout.split('\n')][:-1]
+
+    for index, key in enumerate(keys or ['type']):
+        key, index = (key, index) if sort in ['-s', '-S'] else ('path', -1)
+        sorted_values = presorted.get(key, ['<unset>'] * 3)
+        print(key)
+        values = [line[index] for line in output]
+
+        if sort == '-S':
+            sorted_values = list(reversed(sorted_values))
+
+        assert values == sorted_values
+
+    assert not ret.stderr
+    assert ret.returncode == 0
+
+
+@pytest.mark.parametrize('keys', [None, ['num'], ['str', 'num']])
+@pytest.mark.parametrize('reverse', [True, False])
+def test_natural_sort(
```

Comment:
Yes, I do prefer to add a doc string everywhere.
In this specific case, I would not add what you have proposed, but a more explicit form, mentioning that all values given to assets are always strings that contain numbers. I know this is implicitely stated by "natural sort", but I think this would make the reading of the function slightly easier. :)

Reply:
I have added a docstring for completeness sake. Since [natural sort](https://en.wikipedia.org/wiki/Natural_sort_order) is a common sorting algorithm (used by operating systems for folders as well) and clearly defined, I don't think I should explain what natural sorting is in the docstring.